src/MizarWidgets/SamplingWithFrameTrackWidget.cpp

```diff
@@ -100,4 +113,19 @@ void SamplingWithFrameTrackWidget::OnUpdateButtonClicked() {
   ui_->output_->UpdateReport(std::move(report));
 }
 
+bool SamplingWithFrameTrackWidget::IsDataValid(const orbit_mizar_data::MizarPairedData& data,
+                                               std::string_view data_title) const {
+  const bool is_valid = !data.GetFrameTracks().empty();
+  if (!is_valid) {
+    emit ReportError(absl::StrCat(data_title,
```

Comment:
I am not too sure about the right design here. Should we emit the error report here, or return an ErrorMessageOr and handle it later?

Anyways. If we emit the error here, I would rename the function, to e.g. IsDataValidAndEmitErrorIfNot

Reply:
Good point. Thank you. The function returns ErrorMessageOr now.